Lib/sysconfig.py

```diff
@@ -59,6 +59,21 @@
     }
 
 
+_is_pypy = '__pypy__' in sys.builtin_module_names
+
+# NOTE: site.py has copy of this function.
+# Sync it when modify this function.
+def _get_impllibdir(os_name):
+    if not _is_pypy and os_name != 'nt':
+        return 'python'
+    elif not _is_pypy and os_name == 'nt':
+        return 'Python'
+    elif _is_pypy and os_name != 'nt':
+        return 'pypy'
+    elif _is_pypy and os_name == 'nt':
+        return 'PyPy'
+
```

Comment:
Is there any reason why you have duplicated the code instead of importing them from ``site``? The site module is almost always imported at startup.

```suggestion

# NOTE: site.py has copy of this function.
# Sync it when modify this function.
def _get_impllibdir(os_name):
    if _is_pypy:
        return 'PyPy' if os_name == 'nt' else 'pypy'
    else:
        return 'Python' if os_name == 'nt' else 'python'
```

Reply:
I think the design is that the site module can be overridden by the user, so it cannot be the canonical source of information. I think the sysconfig code was originally copied to prevent site.py importing sysconfig.py during startup, which would slow things down. I was just following that usage pattern.

Reply:
Do you have references for that?  `sitecustomize` is for custom behaviour, but I don’t see that `site` is meant to be replaced.

Reply:
I could see a bootstrap problem though, for example during Python build we may need sysconfig but can’t import site (that needs modules such as io and others).